error in TensorFunctions.h during make on macOS with M2 chip

<!--
Text in these brackets are comments, and won't be visible when you submit your
issue. Please read before submitting.

Before opening an issue, please review the build documentation here

https://github.com/ANTsX/ANTs/wiki/Compiling-ANTs-on-Linux-and-Mac-OS

You can also search the wiki and previous issues using the main Github search
bar. Enter search terms and select "In this repository".

If you are using system ITK or VTK, please verify you have the correct
versions (see the Wiki link above). If you can, please also attempt a
Superbuild and let us know if that compiles successfully.
-->

**When did the error occur?**
<!-- place X in the appropriate box -->
[ ] CMake configuration (cmake / ccmake)
[X] Compilation (make)
[ ] Installation (make install)

**Build environment**
 - OS: MacOS
 - OS version: 14.1
 - Type of system: macbook pro M2

<!--
If you are building inside a virtual machine, container, Cygwin, Windows
Subsystem for Linux, or other non-native environment, please let us know and
include details of both the virtual Linux and the host OS.
-->

**ANTs version**
<!--
Specify the release tag or commit hash of the code you are building
from with `git show` in your source directory. If you downloaded a snapshot as a
ZIP file, please provide the tag version (if applicable) or date of the download.
-->
git clone https://github.com/ANTsX/ANTs.git on Jan 16, 2024

**Build configuration and logs**
<!--
Please attach the following files (relative to your build directory)

  - build.log (the terminal output from make, eg `make | tee build.log`)
  - CMakeCache.txt
  - CMakeFiles/CMakeError.log
  - CMakeFiles/CMakeOutput.log
[CMakeCache.txt](https://github.com/ANTsX/ANTs/files/13956082/CMakeCache.txt)

-->
[build.log](https://github.com/ANTsX/ANTs/files/13956078/build.l
[CMakeOutput.log](https://github.com/ANTsX/ANTs/files/13956091/CMakeOutput.log)
og)
[CMakeError.log](https://github.com/ANTsX/ANTs/files/13956089/CMakeError.log)

[CMakeCache.txt](https://github.com/ANTsX/ANTs/files/13956082/CMakeCache.txt)




**Additional context**
<!--Add any other context about the problem here.-->

Error message:

```
In file included from /Users/xeniagrande/ANTs/Examples/antsRegistration2DDouble.cxx:1:
In file included from /Users/xeniagrande/ANTs/Examples/antsRegistrationTemplateHeader.h:14:
In file included from /Users/xeniagrande/ANTs/Examples/itkantsRegistrationHelper.h:21:
In file included from /Users/xeniagrande/ANTs/Utilities/ReadWriteData.h:26:
In file included from /Users/xeniagrande/ANTs/Tensor/itkLogTensorImageFilter.hIn file included from :/Users/xeniagrande/ANTs/Examples/antsRegistrationTemplateHeader.cxx20::
1:
/Users/xeniagrande/ANTs/Tensor/TensorFunctions.hIn file included from :/Users/xeniagrande/ANTs/Examples/antsRegistrationTemplateHeader.h441::1414:
:In file included from  /Users/xeniagrande/ANTs/Examples/itkantsRegistrationHelper.h:error: no viable conversion from 'vnl_matrix_inverse<double>' to 'MatrixType' (aka 'vnl_matrix<double>')
21:
In file included from /Users/xeniagrande/ANTs/Utilities/ReadWriteData.h:26:
In file included from /Users/xeniagrande/ANTs/Tensor/itkLogTensorImageFilter.h:20:
/Users/xeniagrande/ANTs/Tensor/TensorFunctions.h:441  MatrixType inv = vnl_matrix_inverse<double>(DT);:
14             ^     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~:
 error: no viable conversion from 'vnl_matrix_inverse<double>' to 'MatrixType' (aka 'vnl_matrix<double>')
  MatrixType inv = vnl_matrix_inverse<double>(DT);
             ^     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/xeniagrande/build/staging/include/ITK-5.4/vnl/vnl_matrix.h:141:3: note: candidate constructor not viable: no known conversion from 'vnl_matrix_inverse<double>' to 'const vnl_matrix<double> &' for 1st argument
  vnl_matrix(vnl_matrix<T> const&);    // from another matrix.
  ^
/Users/xeniagrande/build/staging/include/ITK-5.4/vnl/vnl_matrix.h:141:3: note: candidate constructor not viable: no known conversion from 'vnl_matrix_inverse<double>' to 'const vnl_matrix<double> &' for 1st argument
/Users/xeniagrande/build/staging/include/ITK-5.4/vnl/vnl_matrix.h:148:3: note: candidate constructor not viable: no known conversion from 'vnl_matrix_inverse<double>' to 'vnl_matrix<double> &&' for 1st argument
  vnl_matrix(vnl_matrix<T> const&);    // from another matrix.
  ^  vnl_matrix(vnl_matrix<T> &&);

  ^
/Users/xeniagrande/build/staging/include/ITK-5.4/vnl/algo/vnl_matrix_inverse.h/Users/xeniagrande/build/staging/include/ITK-5.4/vnl/vnl_matrix.h:148:3: note: candidate constructor not viable: no known conversion from 'vnl_matrix_inverse<double>' to 'vnl_matrix<double> &&' for 1st argument
:40:  vnl_matrix(vnl_matrix<T> &&);12
:  ^ 
note: explicit conversion function is not a candidate
/Users/xeniagrande/build/staging/include/ITK-5.4/vnl/algo/vnl_matrix_inverse.h:40:12: note: explicit conversion function is not a candidate
  explicit operator vnl_matrix<T>() const { return this->inverse(); }
           ^
  explicit operator vnl_matrix<T>() const { return this->inverse(); }
           ^
In file included from /Users/xeniagrande/ANTs/Examples/antsRegistration2DDouble.cxx:1:
In file included from /Users/xeniagrande/ANTs/Examples/antsRegistrationTemplateHeader.h:14:
In file included from /Users/xeniagrande/ANTs/Examples/itkantsRegistrationHelper.h:21:
In file included from /Users/xeniagrande/ANTs/Utilities/ReadWriteData.h:26:
In file included from /Users/xeniagrande/ANTs/Tensor/itkLogTensorImageFilter.h:20:
/Users/xeniagrande/ANTs/Tensor/TensorFunctions.h:840:14: error: no viable conversion from 'vnl_matrix_inverse<double>' to 'MatrixType' (aka 'vnl_matrix<double>')
  MatrixType inv = vnl_matrix_inverse<double>(DT);
             ^     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/xeniagrande/build/staging/include/ITK-5.4/vnl/vnl_matrix.h:141:3: note: candidate constructor not viable: no known conversion from 'vnl_matrix_inverse<double>' to 'const vnl_matrix<double> &' for 1st argument
  vnl_matrix(vnl_matrix<T> const&);    // from another matrix.
  ^
/Users/xeniagrande/build/staging/include/ITK-5.4/vnl/vnl_matrix.h:148:3: note: candidate constructor not viable: no known conversion from 'vnl_matrix_inverse<double>' to 'vnl_matrix<double> &&' for 1st argument
  vnl_matrix(vnl_matrix<T> &&);
  ^
/Users/xeniagrande/build/staging/include/ITK-5.4/vnl/algo/vnl_matrix_inverse.h:40:12: note: explicit conversion function is not a candidate
  explicit operator vnl_matrix<T>() const { return this->inverse(); }
           ^
In file included from /Users/xeniagrande/ANTs/Examples/antsRegistrationTemplateHeader.cxx:1:
In file included from /Users/xeniagrande/ANTs/Examples/antsRegistrationTemplateHeader.h:14:
In file included from /Users/xeniagrande/ANTs/Examples/itkantsRegistrationHelper.h:21:
In file included from /Users/xeniagrande/ANTs/Utilities/ReadWriteData.h:26:
In file included from /Users/xeniagrande/ANTs/Tensor/itkLogTensorImageFilter.h:20:
/Users/xeniagrande/ANTs/Tensor/TensorFunctions.h:840:14: error: no viable conversion from 'vnl_matrix_inverse<double>' to 'MatrixType' (aka 'vnl_matrix<double>')
  MatrixType inv = vnl_matrix_inverse<double>(DT);
             ^     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/xeniagrande/build/staging/include/ITK-5.4/vnl/vnl_matrix.h:141:3: note: candidate constructor not viable: no known conversion from 'vnl_matrix_inverse<double>' to 'const vnl_matrix<double> &' for 1st argument
  vnl_matrix(vnl_matrix<T> const&);    // from another matrix.
  ^
/Users/xeniagrande/build/staging/include/ITK-5.4/vnl/vnl_matrix.h:148:3: note: candidate constructor not viable: no known conversion from 'vnl_matrix_inverse<double>' to 'vnl_matrix<double> &&' for 1st argument
  vnl_matrix(vnl_matrix<T> &&);
  ^
/Users/xeniagrande/build/staging/include/ITK-5.4/vnl/algo/vnl_matrix_inverse.h:40:12: note: explicit conversion function is not a candidate
  explicit operator vnl_matrix<T>() const { return this->inverse(); }
           ^
```

--- a/Tensor/TensorFunctions.h
+++ b/Tensor/TensorFunctions.h
@@ -438,7 +438,7 @@ GetMetricTensorCost(TVectorType dpath, TTensorType dtv, unsigned int matrixpower
   vec(0, 0) = dpath[0];
   vec(1, 0) = dpath[1];
   vec(2, 0) = dpath[2];
-  MatrixType inv = vnl_matrix_inverse<double>(DT);
+  MatrixType inv = vnl_matrix_inverse<double>(DT).inverse();
   for (unsigned int lo = 1; lo < matrixpower; lo++)
   {
     inv = inv * inv;
@@ -837,7 +837,7 @@ GetMetricTensorCost(itk::Vector<float, 3> dpath, TTensorType dtv)
   vec(0, 0) = dpath[0];
   vec(1, 0) = dpath[1];
   vec(2, 0) = dpath[2];
-  MatrixType inv = vnl_matrix_inverse<double>(DT);
+  MatrixType inv = vnl_matrix_inverse<double>(DT).inverse();
 
   MatrixType sol = vec.transpose() * inv * vec;
   float      cost = sol(0, 0) / etot;

